src/test/kotlin/org/rust/ide/actions/RustStringLiteralJoinLinesHandlerTest.kt

```diff
@@ -0,0 +1,102 @@
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.IdeActions
+import org.intellij.lang.annotations.Language
+import org.rust.lang.RustTestCaseBase
+
+class RustStringLiteralJoinLinesHandlerTest : RustTestCaseBase() {
+    override val dataPath: String = ""
+
+    private fun doTest(
+        @Language("Rust") before: String,
+        @Language("Rust") after: String
+    ) {
+        checkByText("foo.rs", before, after) {
+            myFixture.performEditorAction(IdeActions.ACTION_EDITOR_JOIN_LINES)
+        }
+    }
+
+    fun testNoEscape() = doTest(
+        """
+        fn main() {
+            "Hello<caret>,
+             World"
+        }
+        """,
+        """
+        fn main() {
+            "Hello,<caret> World"
+        }
+        """
+    )
+
+    fun testNewlineEscape() = doTest(
+        """
+        fn main() {
+            "He<caret>llo, \
+             World"
+        }
+        """,
+        """
+        fn main() {
+            "Hello,<caret> World"
+        }
+        """
+    )
+
+    fun testEscapedNewlineEscape() = doTest(
+        """
+        fn main() {
+            "He<caret>llo, \\
+             World"
+        }
+        """,
+        """
+        fn main() {
+            "Hello, \\<caret> World"
+        }
+        """
+    )
+
+    fun testEscapedButNotEscapedInFactNewlineEscape() = doTest(
+        """
+        fn main() {
+            "He<caret>llo, \\\
+             World"
+        }
+        """,
+        """
+        fn main() {
+            "Hello, \\<caret> World"
```

Comment:
Same here :) I think that adding a `\n` is the correct behavior here, but let's not do this for now and wait for the feedback from users?